assets/prototype/application/ui/components/layout/editor-modal/useEditorModalState.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * External dependencies
+ */
+import { useReducer, useEffect } from 'react';
+import { drop, head } from 'ramda';
+
+const CLOSE_ALL = 'CLOSE_ALL';
+const CLOSE_EDITOR = 'CLOSE_EDITOR';
+const OPEN_EDITOR = 'OPEN_EDITOR';
+
+type ActionType = 'CLOSE_ALL' | 'CLOSE_EDITOR' | 'OPEN_EDITOR';
+
+type State = string[];
+
+type Action = {
+	editorId?: string;
+	type: ActionType;
+};
+
+const reducer = (state: State, action: Action) => {
+	const { editorId } = action;
+
+	switch (action.type) {
+		case CLOSE_EDITOR:
+			if (editorId === head(state)) {
```

Comment:
How is it guaranteed that `editorId` will always be the first element of the object?

Reply:
Not sure, this was the previous implementation: https://github.com/eventespresso/event-espresso-core/blob/EDTR/master/assets/ZZZ/data/eventespresso/open-editor-state/index.js#L31

Reply:
Maybe there was a reason for that.
@tn3rb what was the intention there?

Reply:
OK, may be knows the reason